libs/angular-remote-components/src/lib/services/slot.service.ts

```diff
@@ -1,19 +1,81 @@
-import { InjectionToken, Type } from '@angular/core'
-import { Observable } from 'rxjs'
+import { loadRemoteModule } from '@angular-architects/module-federation'
+import { Injectable, InjectionToken, Type } from '@angular/core'
+import { RemoteComponent, RemoteComponentsTopic } from '@onecx/integration-interface'
+import { Observable, map, shareReplay } from 'rxjs'
+import { PermissionService } from './permission.service'
 
 export const SLOT_SERVICE: InjectionToken<SlotService> = new InjectionToken('SLOT_SERVICE')
 
 export type RemoteComponentInfo = { appId: string; productName: string; baseUrl: string }
 
 export type SlotComponentConfiguration = {
-  componentType: Promise<Type<unknown> | undefined> | Type<unknown> | undefined;
-  remoteComponent: RemoteComponentInfo;
-  permissions: Promise<string[]> | string[];
+  componentType: Promise<Type<unknown> | undefined> | Type<unknown> | undefined
+  remoteComponent: RemoteComponentInfo
+  permissions: Promise<string[]> | string[]
 }
 
 export interface SlotService {
   init(): Promise<void>
-  getComponentsForSlot(
-    slotName: string
-  ): Observable<SlotComponentConfiguration[]>
+  getComponentsForSlot(slotName: string): Observable<SlotComponentConfiguration[]>
+  isComponentDefinedForSlot(slotName: string): Observable<boolean>
+}
+
+@Injectable({ providedIn: 'root' })
+export class SlotService implements SlotService {
+  remoteComponents$ = new RemoteComponentsTopic()
+
+  constructor(private permissionsService: PermissionService) {}
+
+  async init(): Promise<void> {
+    return Promise.resolve()
+  }
+
+  getComponentsForSlot(slotName: string): Observable<SlotComponentConfiguration[]> {
+    return this.remoteComponents$.pipe(
+      map((remoteComponentsInfo) =>
+        (remoteComponentsInfo.slots?.find((slotMapping) => slotMapping.name === slotName)?.components ?? [])
+          .map((remoteComponentName) => remoteComponentsInfo.components.find((rc) => rc.name === remoteComponentName))
+          .filter((remoteComponent): remoteComponent is RemoteComponent => !!remoteComponent)
+          .map((remoteComponent) => remoteComponent)
+      ),
+      map((infos) =>
+        infos.map((remoteComponent) => ({
+          componentType: this.loadComponent(remoteComponent),
+          remoteComponent,
+          permissions: this.permissionsService.getPermissions(remoteComponent.appId, remoteComponent.productName),
+        }))
+      ),
+      shareReplay()
+    )
+  }
+
+  isComponentDefinedForSlot(slotName: string): Observable<boolean> {
```

Comment:
better isSomeComponentDefinedForSlot. Otherwise it feels like a function that retuns the info if a specific component is defined for a slot